File: ios/EnumMap.swift
//
//  EnumMap.swift
//  react-native-paypal-checkout
//
//  Created by admin on 2023/3/27.
//

import Foundation
import PayPalCheckout


func getEnvironment(data: String)-> Environment {
    switch(data) {
        case "LIVE":
            return .live
        case "SANDBOX":
            return .sandbox
        default:
            return .stage
    }
}

func getUserAction(data: String?) -> UserAction {
    guard let data = data else { return .payNow }
    switch data {
    case "PAY_NOW":
        return .payNow
    case "CONTINUE":
        return .continue
    default:
        return .payNow
    }
}

func getCurrencyCode(data: String?) -> CurrencyCode {
    guard let data = data else { return CurrencyCode.usd }
    switch data {
    case "AUD":
        return .aud
    case "BRL":
        return .brl
    case "CAD":
        return .cad
    case "CNY":
        return .cny
    case "CZK":
        return .czk
    case "DKK":
        return .dkk
    case "EUR":
        return .eur
    case "HKD":
        return .hkd
    case "HUF":
        return .huf
    case "INR":
        return .inr
    case "ILS":
        return .ils
    case "JPY":
        return .jpy
    case "MYR":
        return .myr
    case "MXN":
        return .mxn
    case "TWD":
        return .twd
    case "NZD":
        return .nzd
    case "NOK":
        return .nok
    case "PHP":
        return .php
    case "PLN":
        return .pln
    case "GBP":
        return .gbp
    case "RUB":
        return .rub
    case "SGD":
        return .sgd
    case "SEK":
        return .sek
    case "CHF":
        return .chf
    case "THB":
        return .thb
    case "USD":
        return .usd
    default:
        return .usd
    }
}

File: android/src/main/java/com/wutiange/reactnativepaypalcheckout/EnumMap.kt
package com.wutiange.reactnativepaypalcheckout

import com.paypal.checkout.config.Environment
import com.paypal.checkout.createorder.CurrencyCode
import com.paypal.checkout.createorder.OrderIntent
import com.paypal.checkout.createorder.UserAction

fun getEnvironment(data: String?): Environment {
    return when (data) {
        "SANDBOX" -> Environment.SANDBOX
        "LIVE" -> Environment.LIVE
        "LOCAL" -> Environment.LOCAL
        else -> Environment.STAGE
    }
}

private fun getUserAction(data: String?): UserAction {
    return when (data) {
        "PAY_NOW" -> UserAction.PAY_NOW
        "CONTINUE" -> UserAction.CONTINUE
        else -> UserAction.PAY_NOW
    }
}

private fun getOrderIntent(data: String?): OrderIntent {
    return when (data) {
        "CAPTURE" -> OrderIntent.CAPTURE
        "AUTHORIZE" -> OrderIntent.AUTHORIZE
        else -> OrderIntent.AUTHORIZE
    }
}

/**
 * 国家支付编码
 */
private fun getCurrencyCode(data: String?): CurrencyCode {
    return when (data) {
        "AUD" -> CurrencyCode.AUD;
        "BRL" -> CurrencyCode.BRL;
        "CAD" -> CurrencyCode.CAD;
        "CNY" -> CurrencyCode.CNY;
        "CZK" -> CurrencyCode.CZK;
        "DKK" -> CurrencyCode.DKK;
        "EUR" -> CurrencyCode.EUR;
        "HKD" -> CurrencyCode.HKD;
        "HUF" -> CurrencyCode.HUF;
        "INR" -> CurrencyCode.INR;
        "ILS" -> CurrencyCode.ILS;
        "JPY" -> CurrencyCode.JPY;
        "MYR" -> CurrencyCode.MYR;
        "MXN" -> CurrencyCode.MXN;
        "TWD" -> CurrencyCode.TWD;
        "NZD" -> CurrencyCode.NZD;
        "NOK" -> CurrencyCode.NOK;
        "PHP" -> CurrencyCode.PHP;
        "PLN" -> CurrencyCode.PLN;
        "GBP" -> CurrencyCode.GBP;
        "RUB" -> CurrencyCode.RUB;
        "SGD" -> CurrencyCode.SGD;
        "SEK" -> CurrencyCode.SEK;
        "CHF" -> CurrencyCode.CHF;
        "THB" -> CurrencyCode.THB;
        "USD" -> CurrencyCode.USD;

        else -> CurrencyCode.USD;
    }
}

fun String.toCurrencyCode(): CurrencyCode {
    return getCurrencyCode(this)
}

fun String.toEnvironment(): Environment {
    return getEnvironment(this)
}

fun String.toUserAction(): UserAction {
    return getUserAction(this)
}

fun String.toOrderIntent(): OrderIntent {
    return getOrderIntent(this)
}
